### ISSUE TITLE
How are WebView's used in this SDK?

### ISSUE THREAD

**Issue by aaboyd** [intercom/intercom-android#336]
## Version info
  - Intercom for Android version: 3.1.2
  - Android OS version: 5.1.1
  - Android Device: Google Nexus 5

## Expected behavior
Display Messenger

## Actual behavior
Crash due to WebView usage.  I am including this in an application that has system level permissions, and usage of WebView's is forbidden in system apps.

## Steps to reproduce - This would be tough to reproduce unless you built the OS yourself.

I am more interested in how and when the WebView is being used.  Is there anyway to avoid using it?   I'd like to find a reliable way to trigger this to open so I can see if there is a way to work around it.  Right now I can't tell exactly where / when it is being used.

**Comment by ronocod** [comment]
Do you have a stack trace for this crash? 

We're testing out the use of WebViews to render some types of content in the Messenger, but it's on an opt-in basis right now. There was a case in older 3.1 versions where some WebView related code was being run for all apps, that should be fixed in the latest version if you update to that.

**Comment by aaboyd** [comment]
Fixed the original description, this happened in version 3.1.2 not 3.1.3 .

I'll append the full stacktrace, but it really isn't a "bug" on intercom's end, this is just an incompatibility with privileged process'.

As soon as you try to create a `WebView` you will get the following ( in a privileged process )
```
Caused by: java.lang.UnsupportedOperationException: For security reasons, WebView is not allowed in privileged processes
```

So, my followup question is, when / how do you plan to use WebView's going forward?  I will need to address this issue in our code to come up with a solution.  I am trying to get an idea of how soon / if I really need to break out intercom into an app that isn't privileged.


```
android.view.InflateException: Binary XML file line #78: Error inflating class android.webkit.WebView
    at android.view.LayoutInflater.createView(LayoutInflater.java:633)
    at com.android.internal.policy.impl.PhoneLayoutInflater.onCreateView(PhoneLayoutInflater.java:55)
    at android.view.LayoutInflater.onCreateView(LayoutInflater.java:682)
    at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:741)
    at android.view.LayoutInflater.rInflate(LayoutInflater.java:806)
    at android.view.LayoutInflater.rInflate(LayoutInflater.java:809)
    at android.view.LayoutInflater.parseInclude(LayoutInflater.java:916)
    at android.view.LayoutInflater.rInflate(LayoutInflater.java:802)
    at android.view.LayoutInflater.inflate(LayoutInflater.java:504)
    at android.view.LayoutInflater.inflate(LayoutInflater.java:414)
    at io.intercom.android.sdk.conversation.ConversationFragment.onCreateView(ConversationFragment.java:223)
    at android.support.v4.app.Fragment.performCreateView(Fragment.java:2192)
    at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1299)
    at android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1528)
    at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1595)
    at android.support.v4.app.BackStackRecord.executeOps(BackStackRecord.java:758)
    at android.support.v4.app.FragmentManagerImpl.executeOps(FragmentManager.java:2363)
    at android.support.v4.app.FragmentManagerImpl.executeOpsTogether(FragmentManager.java:2149)
    at android.support.v4.app.FragmentManagerImpl.optimizeAndExecuteOps(FragmentManager.java:2103)
    at android.support.v4.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:2013)
    at android.support.v4.app.FragmentManagerImpl$1.run(FragmentManager.java:710)
    at android.os.Handler.handleCallback(Handler.java:739)
    at android.os.Handler.dispatchMessage(Handler.java:95)
    at android.os.Looper.loop(Looper.java:135)
    at android.app.ActivityThread.main(ActivityThread.java:5294)
    at java.lang.reflect.Method.invoke(Native Method)
    at java.lang.reflect.Method.invoke(Method.java:372)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:904)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:699)
Caused by: java.lang.reflect.InvocationTargetException
    at java.lang.reflect.Constructor.newInstance(Native Method)
    at java.lang.reflect.Constructor.newInstance(Constructor.java:288)
    at android.view.LayoutInflater.createView(LayoutInflater.java:607)
    at com.android.internal.policy.impl.PhoneLayoutInflater.onCreateView(PhoneLayoutInflater.java:55) 
    at android.view.LayoutInflater.onCreateView(LayoutInflater.java:682) 
    at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:741) 
    at android.view.LayoutInflater.rInflate(LayoutInflater.java:806) 
    at android.view.LayoutInflater.rInflate(LayoutInflater.java:809) 
    at android.view.LayoutInflater.parseInclude(LayoutInflater.java:916) 
    at android.view.LayoutInflater.rInflate(LayoutInflater.java:802) 
    at android.view.LayoutInflater.inflate(LayoutInflater.java:504) 
    at android.view.LayoutInflater.inflate(LayoutInflater.java:414) 
    at io.intercom.android.sdk.conversation.ConversationFragment.onCreateView(ConversationFragment.java:223) 
    at android.support.v4.app.Fragment.performCreateView(Fragment.java:2192) 
    at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1299) 
    at android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1528) 
    at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1595) 
    at android.support.v4.app.BackStackRecord.executeOps(BackStackRecord.java:758) 
    at android.support.v4.app.FragmentManagerImpl.executeOps(FragmentManager.java:2363) 
    at android.support.v4.app.FragmentManagerImpl.executeOpsTogether(FragmentManager.java:2149) 
    at android.support.v4.app.FragmentManagerImpl.optimizeAndExecuteOps(FragmentManager.java:2103) 
    at android.support.v4.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:2013) 
    at android.support.v4.app.FragmentManagerImpl$1.run(FragmentManager.java:710) 
    at android.os.Handler.handleCallback(Handler.java:739) 
    at android.os.Handler.dispatchMessage(Handler.java:95) 
    at android.os.Looper.loop(Looper.java:135) 
    at android.app.ActivityThread.main(ActivityThread.java:5294) 
    at java.lang.reflect.Method.invoke(Native Method) 
    at java.lang.reflect.Method.invoke(Method.java:372) 
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:904) 
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:699) 
Caused by: java.lang.UnsupportedOperationException: For security reasons, WebView is not allowed in privileged processes
    at android.webkit.WebViewFactory.getProvider(WebViewFactory.java:103)
    at android.webkit.WebView.getFactory(WebView.java:2194)
    at android.webkit.WebView.ensureProviderCreated(WebView.java:2189)
    at android.webkit.WebView.setOverScrollMode(WebView.java:2248)
    at android.view.View.<init>(View.java:3597)
    at android.view.View.<init>(View.java:3691)
    at android.view.ViewGroup.<init>(ViewGroup.java:498)
    at android.widget.AbsoluteLayout.<init>(AbsoluteLayout.java:55)
    at android.webkit.WebView.<init>(WebView.java:544)
    at android.webkit.WebView.<init>(WebView.java:489)
    at android.webkit.WebView.<init>(WebView.java:472)
    at android.webkit.WebView.<init>(WebView.java:459)
    at java.lang.reflect.Constructor.newInstance(Native Method) 
    at java.lang.reflect.Constructor.newInstance(Constructor.java:288) 
    at android.view.LayoutInflater.createView(LayoutInflater.java:607) 
    at com.android.internal.policy.impl.PhoneLayoutInflater.onCreateView(PhoneLayoutInflater.java:55) 
    at android.view.LayoutInflater.onCreateView(LayoutInflater.java:682) 
    at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:741) 
    at android.view.LayoutInflater.rInflate(LayoutInflater.java:806) 
    at android.view.LayoutInflater.rInflate(LayoutInflater.java:809) 
    at android.view.LayoutInflater.parseInclude(LayoutInflater.java:916) 
    at android.view.LayoutInflater.rInflate(LayoutInflater.java:802) 
    at android.view.LayoutInflater.inflate(LayoutInflater.java:504) 
    at android.view.LayoutInflater.inflate(LayoutInflater.java:414) 
    at io.intercom.android.sdk.conversation.ConversationFragment.onCreateView(ConversationFragment.java:223) 
    at android.support.v4.app.Fragment.performCreateView(Fragment.java:2192) 
    at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1299) 
    at android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1528) 
    at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1595) 
    at android.support.v4.app.BackStackRecord.executeOps(BackStackRecord.java:758) 
    at android.support.v4.app.FragmentManagerImpl.executeOps(FragmentManager.java:2363) 
    at android.support.v4.app.FragmentManagerImpl.executeOpsTogether(FragmentManager.java:2149) 
    at android.support.v4.app.FragmentManagerImpl.optimizeAndExecuteOps(FragmentManager.java:2103) 
    at android.support.v4.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:2013) 
    at android.support.v4.app.FragmentManagerImpl$1.run(FragmentManager.java:710) 
    at android.os.Handler.handleCallback(Handler.java:739) 
    at android.os.Handler.dispatchMessage(Handler.java:95) 
    at android.os.Looper.loop(Looper.java:135) 
    at android.app.ActivityThread.main(ActivityThread.java:5294) 
    at java.lang.reflect.Method.invoke(Native Method) 
    at java.lang.reflect.Method.invoke(Method.java:372) 
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:904) 
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:699) 
```

**Comment by ronocod** [comment]
The instantiating of the WebView on apps that aren't opted in was a bug in 3.1.2, it's fixed on 3.1.3 and above.

Currently we're just investigating the usage of WebViews for some content rendering. We don't have any plans to enable this for all apps in the near future, and if we do enable this globally it will only be on SDK versions that handle any situations like this where WebViews can't be used.
